modules/ROOT/pages/reliability-patterns.adoc

```diff
@@ -0,0 +1,162 @@
+----
+<!-- This is the reliable acquisition flow in the reliability pattern.  -->
+<flow name="reliable-data-acquisition">
+	<http:listener config-ref="HTTP_Listener_config" path="transactionalEndpoint"/>
+	<vm:publish config-ref="VM_Config" queueName="toTransactionalVM"/> <!--1-->
+</flow>
+
+<!-- This is the application logic flow in the reliability pattern.
+	It is a wrapper around a sub-flow, "business-logic-processing". -->
+<flow name="main-flow">
+	<vm:listener doc:name="Listener" config-ref="VM_Config" queueName="toTransactionalVM"
+		transactionalAction="ALWAYS_BEGIN"/> <!--2-->
+	<flow-ref name="business-logic-processing"/>
+</flow>
+
+<!-- This is where the real work of the application will happen. -->
+<sub-flow name="business-logic-processing">
+	<!--
+		This flow is where the actual business-logic is performed.
+	-->
+</sub-flow>
+----
+<1> The message is written to the VM queue. It is now available for processing by the main flow.
+<2> The message is read from the VM queue transactionally. This ensures that if an error occurs, the reading will be rolled back and the message reprocessed.
+
+
+== Implementing a Reliable Acquisition Flow
+
+The example provided in Implementing a Reliability Pattern showed a reliable acquisition flow for an HTTP listener to a VM publish-operation. Now, let's focus on a reliable acquisition flow that has a non-transactional Message Source: FTP-to-VM (Or SFTP, FTPS).
+
+In this scenario, we are working with a resource-based connection. When using resource-based connections, you need to ensure to read the resource only once. This is done by setting `watermarkEnabled` to `true`.
+
+You must not set the `auto-delete` parameter to `true` because doing so will delete the file as soon as the flow finishes, even if the processing is done by another flow. The same applies to `moveToDirectory`.
+
+=== Example: FTP to VM
+
+The following code implements a reliable acquisition flow from an FTP listener (On New or Updated File) to a VM queue by a publish operation:
+
+[source,xml,linenums]
+----
+<ftp:config name="FTP_Config">
+	<ftp:connection host="localhost" username="max" password="mulesoft"/>
+</ftp:config>
+
+<flow name="reliable-data-acquisition">
+	<ftp:listener config-ref="FTP_Config" watermarkEnabled="true" directory="someDirectory/in"> <!--1-->
+		<scheduling-strategy >
+			<fixed-frequency />
+		</scheduling-strategy>
+	</ftp:listener>
+	<vm:publish config-ref="VM_Config" queueName="toTransactionalVM"/>
+</flow>
+
+<flow name="doc-examplesFlow">
+	<vm:listener queueName="toTransactionalVM" config-ref="VM_Config" transactionalAction="ALWAYS_BEGIN"/>
+	<set-variable value="#[attributes.filename]" variableName="filepath"/>
+	<!-- File content is already present in payload -->
+	<flow-ref name="business-logic-processing"/>
+	<ftp:delete path="#[vars.filepath]"/> <!--2-->
+</flow>
+----
+<1> In case you wish to perform a transformation before publishing to VM, you may add a max redelivery count.
```

Comment:
To perform a transformation before publishing to VM, add a max redelivery count.